incubator/java-openliberty/image/project/Dockerfile

```diff
@@ -12,23 +12,30 @@ RUN cd /project && mvn -B install dependency:go-offline -DskipTests
 COPY ./util /project/util
 RUN  /project/util/check_version build
 
-# Copy the user's application pom and resolve all dependencies, also trigger Open Liberty install
+# Copy the user's application pom and resolve all dependencies
 COPY ./user-app/pom.xml /project/user-app/pom.xml
+RUN cd /project/user-app && mvn package -Dskip=true -DskipTests
+
+# Download Openliberty here to prevent a re-download
+RUN cd /project/user-app && mvn -B liberty:install-server liberty:install-feature liberty:create
+
+# validate the config.
 COPY ./validate.sh /project/user-app/validate.sh
 RUN cd /project/user-app && ./validate.sh
 
 # Copy and build the application source, and run unit tests
 COPY ./user-app/src /project/user-app/src
-RUN cd /project/user-app && mvn -B liberty:create package
+RUN cd /project/user-app && mvn package liberty:deploy -DskipTests
```

Comment:
I wonder if we should do `mvn clean package` if we've already done `mvn package` (i.e. don't assume it's "idempotent" across all builds).

Reply:
Why do we need `liberty:deploy`?

Reply:
The liberty:deploy was my (incorrect) suggestion, but as you pointed out, we don't need to install the app.